Automattic/woocommerce-payments
# feat: display error on invalid bank statement

Fixes https://github.com/Automattic/woocommerce-payments/issues/2380

#### Changes proposed in this Pull Request









I am adding a `Notice` component to display the message.
Ideally, I'd like the message to be displayed closer to the input, but it doesn't look like the Gutenberg components allow that.
Feel free to check out the props: https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/text-control

The validation is performed server-side, so that we don't need to duplicate it on the client.
I added the error message retrieved to the redux store.

I thought this implementation would get us closer to the solution with the least amount of changes.

Let me know if you'd rather have me clear the error message on submit instead. I thought that clearing the error message on text enter would have been more immediate. I know there are several schools of thought on this, including removing the error message _only_ after the input is valid. But that would mean that we would need to duplicate the check on the client.

#### Testing instructions





- Ensure you have the "grouped settings" flag enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- Clear the input for the bank statement
- Submit the data with the "save" button
- A notice should appear to recommend updating values
- Enter new data in the field
- The notice should be removed

![2021-07-08 13 26 00](https://user-images.githubusercontent.com/273592/124972515-14db6600-dff0-11eb-9c13-4d0c2a45187f.gif)


-------------------
